docs/en/enterprise-edition/policy-reference/supply-chain-policies/terraform-policies/ensure-terraform-module-sources-use-tag.adoc

```diff
@@ -0,0 +1,67 @@
+== Terraform module sources do not use a git url with a tag or commit hash revision
+
+
+=== Policy Details 
+
+[width=45%]
+[cols="1,1"]
+|=== 
+|Prisma Cloud Policy ID 
+| TBD
+
+|Checkov ID 
+| https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/module/generic/RevisionVersionTag.py[CKV_TF_2]
+
+|Severity
+|HIGH
+
+|Subtype
+|Build
+
+|Frameworks
+|Terraform
+
+|=== 
+
+
+
+=== Description
+
+Terraform modules are a collection of multiple resource configuration to offer an easy way of repeatable and reusable code logic.
+The most common way is to consume them through the public Terraform registry, which are connected to a VCS, like GitHub.
+This approach is problematic, because the module versions are not immutable and the module can be changed without changing the version, which makes the code vulnerable to a Supply Chain Attack.
+Therefore it is recommended to leverage Git URLs with a commit hash revision to guarantee immutability and consistency. Git tags can be used as well, but are not as immutable as a hash and therefore acceptable, but less preferred. This policy is a less restrictive version of the https://docs.prismacloud.io/en/enterprise-edition/policy-reference/supply-chain-policies/terraform-policies/ensure-terraform-module-sources-use-git-url-with-commit-hash-revision[Terraform module sources do not use a git url with a commit hash revision] policy that only allows for hashes.
```

Comment:
To mitigate this risk, it is recommended to use Git URLs with a commit hash revision to guarantee immutability and consistency. Although Git tags can also be used, they are not as immutable as a hash and are therefore less preferred.